runtime/shared_common/CompositeCache.cpp

```diff
@@ -3798,20 +3802,24 @@ SH_CompositeCacheImpl::isAddressInMetaDataArea(const void* address) const
 
 /**
  * Check whether given address is in the Shared Classes cache
- * Note - does not include the cache header
  *
  * @param [in] address  Address of cache item
+ * @param [in] includeHeaderReadWriteArea  Whether to check cache header and read write area.
  *
  * @return true if address is within the shared class cache boundaries, false otherwise
  */
 bool
-SH_CompositeCacheImpl::isAddressInCache(const void* address)
+SH_CompositeCacheImpl::isAddressInCache(const void* address, bool includeHeaderReadWriteArea)
 {
 	if (!_started) {
 		Trc_SHR_Assert_ShouldNeverHappen();
 		return false;
 	}
-	return ((address >= CASTART(_theca)) && (address <= getCacheLastEffectiveAddress()));
+	if (includeHeaderReadWriteArea) {
+		return (((address >= _theca)) && (address <= CAEND(_theca)));
+	} else {
+		return (((address >= CASTART(_theca))) && (address <= CAEND(_theca)));
```

Comment:
Too many parentheses in the return statements; this one should be:
```
	return (address >= CASTART(_theca)) && (address <= CAEND(_theca));
```

Reply:
Removed extra parentheses